exporter/honeycombmarkerexporter/config.go

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package honeycombexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/honeycombexporter"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+// Config defines configuration for the Honeycomb Marker exporter.
+type Config struct {
+	// APIKey is the authentication token associated with the Honeycomb account.
+	APIKey string `mapstructure:"api_key"`
+
+	// API URL to use (defaults to https://api.honeycomb.io)
+	APIURL string `mapstructure:"api_url"`
+
+	// Markers is the list of markers to create
+	Markers []marker `mapstructure:"markers"`
+}
+
+type marker struct {
+	// MarkerType defines the type of marker.  Markers with the same type appear in Honeycomb with the same color
+	MarkerType string `mapstructure:"type"`
+
+	// MarkerColor is the color of the marker. Will only be used if the MarkerType does not already exist.
+	MarkerColor string `mapstructure:"color"`
+
+	// MessageField is the attribute that will be used as the message.
+	// If necessary the value will be converted to a string.
+	MessageField string `mapstructure:"message_field"`
+
+	// UrlField is the attribute that will be used as the url.
+	// If necessary the value will be converted to a string.
+	UrlField string `mapstructure:"url_field"`
+
+	// Rules are the OTTL rules that determine when a piece of telemetry should be turned into a Marker
+	Rules Rules `mapstructure:"rules"`
+}
+
+type Rules struct {
+	// ResourceConditions is the list of ottlresource conditions that determine a match
+	ResourceConditions []string `mapstructure:"resource_conditions"`
+
+	// LogConditions is the list of ottllog conditions that determine a match
+	LogConditions []string `mapstructure:"log_conditions"`
+}
+
+func (cfg *Config) Validate() error {
```

Comment:
Lets add another validation check that all Markers have at least 1 rule, and that at least 1 marker is supplied. 

We should also run all the OTTL statements through a parser in this function.  We do that in the transformprocessor here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c506b4e39b9dbbb1ccd6458474d175a9d9455020/processor/transformprocessor/config.go#L34

Reply:
So, I needed a bunch of internal functions for this parsing to be possible. I just duplicated them out of the `transformprocessor/internal` directory, but let me know if there's a better way to access those functions.

Reply:
Ya I shouldve pointed you to the filterprocessor and how it uses filterottl.  We should use that to handle parsing our statements.